src/lib/trace/lttng/providers.h

```diff
@@ -0,0 +1,17 @@
+/* Copyright (c) 2020, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * \file providers.h
```

Comment:
I'd like to avoid a single all-powerful providers.h header; subsystems should just include their own if they need it, right?

Reply:
The way I did this is that C file include `lib/trace/events.h` alone and then they have access to all providers (for the case of LTTng). Else, we would have to add another layer that has per subsystem events like `lib/trace/events_circuit.h` type of files.

No strong opinion, whatever you prefer. I went for ease of use?